### DISCUSSION TITLE
Micronaut and Metrics

### DISCUSSION THREAD

**Discussion by cemo** [open-telemetry/opentelemetry-java-instrumentation#3774] [Q&A]
Hi, I am trying to understand the process of how metrics are working. Since currently I see no metrics on our micronaut based application, I am trying to figure out how things are working. The libraries we are using are supported by opentelemetry as well. I can not understand why metrics are not available.

Can someone enlight me please about it?

**Comment by iNikem** [comment]
We currently only produce traces, not metrics, from instrumented libraries. Because OpenTelemetry Java API/SDK does not provide yet support for metrics. Because metrics part of the specification is not stabilised yet.

**Reply by cemo** [reply]
@iNikem I see some metrics such as `runtime_jvm_memory_pool` on metrics endpoint. What are these numbers?

**Reply by iNikem** [reply]
I am actually not sure. @mateuszrzeszutek do you know anything about this?

**Reply by mateuszrzeszutek** [reply]
We do produce some metrics, like the [runtime metrics](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/MemoryPools.java) for example. We haven't been actively working on adding more metrics (to popular HTTP frameworks for example) because of the API/SDK unstability, and until it becomes stable those few runtime metrics will probably be all you can observe.

**Reply by cemo** [reply]
Hi, I started to see some progress on towards on metrics implementations. Where can I see its documentation and existing metrics per libraries?

**Reply by mateuszrzeszutek** [reply]
Hey @cemo ,
Right now we still don't have that many metrics instrumentations, the work on that has pretty much just started recently. We capture [HTTP metrics](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md) in all HTTP client & server instrumentations, and a couple of JVM metrics for which we don't have any docs - they're bound to change anyway soon.
